lib/THCUNN/generic/SpatialAveragePooling.cu

```diff
@@ -27,11 +27,6 @@ static inline void THNN_(SpatialAveragePooling_shapeCheck)(
 
   THCUNN_argCheck(state, ndim == 3 || ndim == 4, 2, input,
                   "3D or 4D input tensor expected but got: %s");
-
```

Comment:
I think this PR is good.
There was a copy/paste problem in https://github.com/torch/cunn/commit/b923c23d4c058e3c4b4784a6c67752967ab8bbf3#diff-bd0935145314cc2aeb3c96b97076eb80L40 that introduced the shapeCheck problem.

BTW, there is still a reminiscent extra shapeCheck in the C counterpart https://github.com/torch/nn/blob/master/lib/THNN/generic/SpatialAveragePooling.c#L136-L139